### PR TITLE
[FIX] l10n_latam_invoice_document: Allow to change the l10n_latam_document…

### PR DESCRIPTION
**Pull request by suv-bmya**
…_number manually when the invoice is in draft status

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by suv-bmya** [comment]
@jco-odoo here the first approach to allow change the l10n_latam_document_number manually. 

I removed the condition related with the l10n_latam_manual_document_number and highest_name from the model and it works well. On the other hand, I've removed the same fields from the view but I had to remove the l10n_latam_manual_document_number too because l10n_latam_manual_document_number is set to False once the first invoice is created.
Finally,  I also removed the posted_before field to readonly attribute but I'm not sure if we want to allow change the document number once the invoice is posted.

Take a look, please. I would like discuss it tomorrow in the meeting.

File: addons/l10n_latam_invoice_document/views/account_move_view.xml [hunk 1]
```diff
@@ -45,7 +45,7 @@
                     attrs="{'invisible': [('l10n_latam_use_documents', '=', False)], 'required': [('l10n_latam_use_documents', '=', True)], 'readonly': [('posted_before', '=', True)]}"
                     domain="[('id', 'in', l10n_latam_available_document_type_ids)]" options="{'no_open': True, 'no_create': True}"/>
                 <field name="l10n_latam_document_number"
-                    attrs="{'invisible': ['|', ('l10n_latam_use_documents', '=', False), ('l10n_latam_manual_document_number', '=', False), '|', '|', ('l10n_latam_use_documents', '=', False), ('highest_name', '!=', False), ('state', '!=', 'draft')], 'required': ['|', ('l10n_latam_manual_document_number', '=', True), ('highest_name', '=', False)], 'readonly': [('posted_before', '=', True), ('state', '!=', 'draft')]}"/>
+                    attrs="{'invisible': [('l10n_latam_use_documents', '=', False)], 'required': ['|', ('l10n_latam_manual_document_number', '=', True), ('highest_name', '=', False)]}"/>
```
Comment (jco-odoo) [inline addons/l10n_latam_invoice_document/views/account_move_view.xml:48]: @suv-bmya It does need to stay read-only in some cases.  This is quite a rough change

**Comment by vbe-odoo** [comment]
@vbe-odoo do FUN test on this one

**Comment by vbe-odoo** [comment]
Hi @suv-bmya @Danisan I am a bit lost on this one, maybe you can help me out.
I am testing it and I can only change the l10n_latam_document_number on the Vendor Bill side, but not on the Invoice Side.
Should this happen on both sides?
In both cases, I have created an Invoice/Vendor Bill first, and then on the 2nd, 3rd document created tried to see if the l10n_latam_document_number was visible and editable. And this only happened on the Vendor Bill side.

Thanks!
best
@ren-odoo

**Comment by suv-bmya** [comment]
> Hi @suv-bmya @Danisan I am a bit lost on this one, maybe you can help me out.
> I am testing it and I can only change the l10n_latam_document_number on the Vendor Bill side, but not on the Invoice Side.
> Should this happen on both sides?
> In both cases, I have created an Invoice/Vendor Bill first, and then on the 2nd, 3rd document created tried to see if the l10n_latam_document_number was visible and editable. And this only happened on the Vendor Bill side.
> 
> Thanks!
> best
> @ren-odoo

@vbe-odoo , What runbot did you use to test it?

**Comment by vbe-odoo** [comment]
Hi @suv-bmya I've used the one attahed to this PR. Details below:
Runbot database: https://8899933-14-0-all.runbot58.odoo.com/
Runbot branch & Builds: https://runbot.odoo.com/runbot/r-d-1?search=bmya%3A14.0_allow_change_l10n_latam_document_number

Thanks!

**Comment by suv-bmya** [comment]
> Hi @suv-bmya I've used the one attached to this PR. Details below:
> Runbot database: https://8899933-14-0-all.runbot58.odoo.com/
> Runbot branch & Builds: https://runbot.odoo.com/runbot/r-d-1?search=bmya%3A14.0_allow_change_l10n_latam_document_number
> 
> Thanks!

Thanks @vbe-odoo . 

To change the document number you should confirm the invoice. The flow is:
* Create draft invoice
* Confirm invoice: the document number is assigned automatically
* Set the invoice to draft: the document number is visible and you can edit it

image

**Comment by vbe-odoo** [comment]
Hey, @suv-bmya Oh I see! I was missing that step, the reset to draft.
I tested for the CL company and it's working OK.
I also installed AR just in case and tested it, and didn't affect how AR is working, so everything seems OK.

Just to keep you in the loop @jjscarafia this PR is to be able to change the sequence in customer invoices in CL.

This is ready for technical review @jco-odoo 
Thanks!

**Comment by jjscarafia** [comment]
Thanks @vbe-odoo !

File: addons/l10n_cl/models/account_move.py [hunk 2]
```diff
@@ -124,3 +124,19 @@ def _get_name_invoice_report(self):
         if self.l10n_latam_use_documents and self.company_id.country_id.code == 'CL':
             return 'l10n_cl.report_invoice_document'
         return super()._get_name_invoice_report()
+
+    @api.onchange('l10n_latam_document_type_id', 'l10n_latam_document_number')
+    def _inverse_l10n_latam_document_number(self):
```
Comment (jco-odoo) [inline addons/l10n_cl/models/account_move.py:129]: After a rebase it should work without overriding this method. @suv-bmya

File: addons/l10n_cl/views/account_move_view.xml [hunk 3]
```diff
@@ -15,6 +15,31 @@
...
+                    	     ('l10n_latam_manual_document_number', '=', False),
+                    	'|', '|', ('l10n_latam_use_documents', '=', False),
+                    	          ('highest_name', '!=', False),
+                    	          ('state', '!=', 'draft'),
+                    	'|', '|', '|', ('l10n_latam_use_documents', '=', False),
+                    	               ('posted_before', '=', False),
+                    	               ('state', '!=', 'draft'),
+                    	               ('country_code', '!=', 'CL')
+                    ],
+                    'readonly': [('posted_before', '=', True), ('state', '!=', 'draft'), ('country_code', '!=', 'CL')]
```
Comment (jco-odoo) [inline addons/l10n_cl/views/account_move_view.xml:36]: As the readonly is True only when the state is not draft, I don't know why we add the extra condition country_code != 'CL'

Also, why do we remove the "required" one or is it still kept magically?

File: addons/l10n_cl/views/account_move_view.xml [hunk 4]
```diff
@@ -15,6 +15,31 @@
...
     </record>
 
+    <record id="view_latam_form_inherit_l10n_cl" model="ir.ui.view">
+        <field name="name">account.move.latam.form.inherit.l10n.cl</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="l10n_latam_invoice_document.view_move_form"/>
+        <field name="arch" type="xml">
+            <field name="l10n_latam_document_number" position="attributes">
+                <attribute name="attrs">
+                    {'invisible': [
```
Comment (jco-odoo) [inline addons/l10n_cl/views/account_move_view.xml:25]: invisible parameters seem correct, but the indentation of the first part can be improved.

**Review by jco-odoo — CHANGES_REQUESTED** [review]
@suv-bmya Should not take you a lot of to finish this.

File: addons/l10n_cl/views/account_move_view.xml (hunk 4 above)
Comment (suv-bmya) [reply addons/l10n_cl/views/account_move_view.xml:25]: it's weird 🤔

File: addons/l10n_cl/models/account_move.py (hunk 2 above)
Comment (suv-bmya) [reply addons/l10n_cl/models/account_move.py:129]: 👍

File: addons/l10n_cl/views/account_move_view.xml (hunk 3 above)
Comment (suv-bmya) [reply addons/l10n_cl/views/account_move_view.xml]: We cannot remove the country_code condition due to the posted_before field.

Ups! I forgot the required attribute 🤓

**Comment by suv-bmya** [comment]
> @suv-bmya Should not take you a lot of to finish this.

@jco-odoo . Done!

File: addons/l10n_cl/views/account_move_view.xml (hunk 3 above)
Comment (jco-odoo) [reply addons/l10n_cl/views/account_move_view.xml]: @suv-bmya I don't understand.  In order for it to be readonly you should not be in draft state anyways.  I think you should only modify it when the state is in draft anyways, so I don't see why you need the extra condition.

Comment (suv-bmya) [reply addons/l10n_cl/views/account_move_view.xml]: Nothing! You're right...We don't need this condition.

**Comment by vbe-odoo** [comment]
@vbe-odoo do a FUN test and confirm

**Comment by vbe-odoo** [comment]
@jco-odoo FUN test done, and LGTH.
Ready for technical review > merge.
Thanks.
cc' @suv-bmya @Danisan

**Comment by jco-odoo** [comment]
@suv-bmya Could you put a standard format commit message: [FIX] l10n_latam_invoice_document: ....

**Comment by suv-bmya** [comment]
> @suv-bmya Could you put a standard format commit message: [FIX] l10n_latam_invoice_document: ....

ups! Sure.

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**